### PR TITLE
libbeat/common/encoding/xml: ignore encoding declarations in proc inst

### LINKED ISSUE #31395: decode_xml and decode_xml_wineventlog fail on UTF-16 charsets
I'm trying to parse a field that contains XML and it logs the following error:

`failed in decode_xml on the "winlog.event_data.param2" field: error decoding XML field: xml: encoding "utf-16" declared but Decoder.CharsetReader is nil`

The field itself is formatted as such:
```
<?xml version="1.0" encoding="utf-16"?>
<AuditBase xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="FreshCredentialAudit">
  snip
  </ContextComponents>
</AuditBase>
```

This was reported, but closed due to no responses, in February here: https://discuss.elastic.co/t/decode-xml-fails-on-utf-16-formatted-input/295996

When on logstash, the filter worked fine, but since moving to elastic agent (until 8.2 when Logstash output may be available) this means most of the data in ADFS authentication logs, for example, isn't available as the user, IP, user agent, and other relevant info is all in the XML.

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This change allows the xml decoder to attempt to decode text that claims to be other
than UTF-8. This allows us to decode documents that have been transformed from other
encodings but have not had their encoding declaration updated to match.

In the case that the text is not UTF-8 we only change the behaviour slightly by
altering the error message from e.g. 'encoding "utf-16" declared but
Decoder.CharsetReader is nil' to e.g. 'XML syntax error on line 1: invalid UTF-8'.

## Why is it important?

Currently we reject text that claims to be another encoding even if it is actually UTF-8. We will still fail on non-UTF-8 text, but with a slightly different error.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes #31395

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -39,6 +39,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 
 - Fix group write permissions on runtime directories. {pull}30869[30869]
 - Store syslog version as string. {pull}31446[31446]
+- Accept XML that declares non-UTF-8 encoding. {issue}31395[31395] {pull}[]
```
Comment (efd6) [inline CHANGELOG.next.asciidoc:42]:
```suggestion
- Accept XML that declares non-UTF-8 encoding. {issue}31395[31395] {pull}31546[31546]
```

File: CHANGELOG.next.asciidoc [hunk 2]
```diff
@@ -39,6 +39,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...main[Check the HEAD dif
 
 - Fix group write permissions on runtime directories. {pull}30869[30869]
 - Store syslog version as string. {pull}31446[31446]
+- Accept XML that declares non-UTF-8 encoding. {issue}31395[31395] {pull}31546[31546]
```
Comment (andrewkroh) [inline CHANGELOG.next.asciidoc:42]: Could you mention decode_xml / decode_xml_wineventlog to give it some more context.

**Review by andrewkroh — APPROVED** [review]

File: CHANGELOG.next.asciidoc (hunk 2 above)
Comment (efd6) [inline CHANGELOG.next.asciidoc:42]: Does this work for you?

```suggestion
- Accept XML that declares non-UTF-8 encoding to allow decode_xml and decode_xml_wineventlog decoding of incorrectly annotated documents. {issue}31395[31395] {pull}31546[31546]
```

Comment (andrewkroh) [reply CHANGELOG.next.asciidoc:42]: SGTM

**Pull request merged**